lib/collection/src/update_workers/flush_workers.rs

```diff
@@ -68,17 +68,8 @@ impl UpdateWorkers {
             Err(err) => {
                 // Since Self::flush_segments is flushing asynchronously, we can get the error
                 // from the previous flush cycle, not necessarily this one.
-                match err {
-                    OperationError::Cancelled { description: _ } => {
-                        log::debug!("Flush was cancelled: {err}");
-                        // Don't report as optimizer error here, since cancellation can happen due to
-                        // trying to flush an already dropped segment
-                    }
-                    _ => {
-                        log::error!("Failed to flush: {err}");
-                        segments.write().report_optimizer_error(err);
-                    }
-                }
+                log::error!("Failed to flush: {err}");
+                segments.write().report_optimizer_error(err);
```

Comment:
This rolls back previous changes, as the cancellation is handled at segment level